storage-quickstart.go

```diff
@@ -32,114 +29,101 @@ func randomString() string {
-
-	// Create a random string for the quick start container
-	containerName := fmt.Sprintf("quickstart-%s", randomString())
 
-	// From the Azure portal, get your storage account blob service URL endpoint.
-	URL, _ := url.Parse(
-		fmt.Sprintf("https://%s.blob.core.windows.net/%s", accountName, containerName))
-
-	// Create a ContainerURL object that wraps the container URL and a request
-	// pipeline to make requests.
-	containerURL := azblob.NewContainerURL(*URL, p)
+	serviceClient, err := azblob.NewServiceClient(url, credential, nil)
+	if err != nil {
+		log.Fatal("Invalid credentials with error: " + err.Error())
+	}
 
 	// Create the container
+	containerName := fmt.Sprintf("quickstart-%s", randomString())
 	fmt.Printf("Creating a container named %s\n", containerName)
-	ctx := context.Background() // This example uses a never-expiring context
-	_, err = containerURL.Create(ctx, azblob.Metadata{}, azblob.PublicAccessNone)
-	handleErrors(err)
+	containerClient := serviceClient.NewContainerClient(containerName)
+	_, err = containerClient.Create(ctx, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
 
-	// Create a file to test the upload and download.
 	fmt.Printf("Creating a dummy file to test the upload and download\n")
-	data := []byte("hello world this is a blob\n")
-	fileName := randomString()
-	err = ioutil.WriteFile(fileName, data, 0700)
-	handleErrors(err)
-
-	// Here's how to upload a blob.
-	blobURL := containerURL.NewBlockBlobURL(fileName)
-	file, err := os.Open(fileName)
-	handleErrors(err)
-
-	// You can use the low-level PutBlob API to upload files. Low-level APIs are simple wrappers for the Azure Storage REST APIs.
-	// Note that PutBlob can upload up to 256MB data in one shot. Details: https://docs.microsoft.com/en-us/rest/api/storageservices/put-blob
-	// Following is commented out intentionally because we will instead use UploadFileToBlockBlob API to upload the blob
-	// _, err = blobURL.PutBlob(ctx, file, azblob.BlobHTTPHeaders{}, azblob.Metadata{}, azblob.BlobAccessConditions{})
-	// handleErrors(err)
-
-	// The high-level API UploadFileToBlockBlob function uploads blocks in parallel for optimal performance, and can handle large files as well.
-	// This function calls PutBlock/PutBlockList for files larger 256 MBs, and calls PutBlob for any file smaller
-	fmt.Printf("Uploading the file with blob name: %s\n", fileName)
-	_, err = azblob.UploadFileToBlockBlob(ctx, file, blobURL, azblob.UploadToBlockBlobOptions{
-		BlockSize:   4 * 1024 * 1024,
-		Parallelism: 16})
-	handleErrors(err)
-
-	// List the container that we have created above
+
+	data := []byte("\nhello world this is a blob\n")
+	blobName := "blob" + "-" + randomString()
+
+	var blockOptions azblob.HighLevelUploadToBlockBlobOption
+
+	blobClient, err := azblob.NewBlockBlobClient(url+containerName+"/"+blobName, credential, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Upload to data to blob storage
+	_, err = blobClient.UploadBufferToBlockBlob(ctx, data, blockOptions)
```

Comment:
It's typical for callers to use a struct literal instead of defining `blockOptions` separately.  So remove the declaration on line 64 and do this.
```go
_, err = blobClient.UploadBufferToBlockBlob(ctx, data, azblob.HighLevelUploadToBlockBlobOption{})
```

Reply:
So that's how you do that, thanks!